### DISCUSSION TITLE
How can I prevent existing formats from being deleted when rendering a different format? (RStudio)

### DISCUSSION THREAD

**Discussion by fretwurst** [quarto-dev/quarto-cli#6614] [Q&A] [answered]
### Description

What I find great about Quarto is that you can create report web pages, and simultaneously create PDF versions or other formats for printing. However, these formats sometimes require specific adjustments (such as templates in LaTeX or CSS in HTML). For instance, if I have finished customizing the PDF version and then only adjust and render the HTML version, the PDF version gets deleted first in the output directory. This means that I always must render all formats so that they are present in the output directory, even though only one format has been modified. This naturally takes a considerable amount of time. Is there a way to configure Quarto to not delete existing formats, as it used to work with bookdown?

**Comment by cderv** [comment]
Could you provide a reproducible example ? 

Currently I can't reproduce. Try 
````markdown
---
title: "Test"
format: 
  pdf: default
  html: default
---

# Minimal example

This is a minimal example
````

* `quarto render index.qmd --to pdf` creates a pdf file
* `quarto render index.qmd --to html` creates a .html file and `index_files` folder **without removing the pdf file**

So are you in a specific context ?  Quarto project ? Latest quarto version ? Sharing more details will help

**Comment by fretwurst** [comment]
Thank you! And yes, it is a quarto-project with Quarto 1.4.322. I have the _quarto.yml: 
```
project:
  title: "formats"
  type: book
  output-dir: report

book:
  chapters:
    - index.qmd
  downloads: [pdf]

format:
  pdf:
    documentclass: scrreprt
  html:
    theme: cosmo
```

And an index file: 

```
---
title: "formats"
---

# Minimal example

This is a minimal example
```

After "Render Book" in the "Build"-Panel from RStudio, all formats are rendered in the "report"-folder. When I use the drop down button from "Render Book" and only choose the "HTML Format", everything in the output-dir "report"-folder is deleted and only the HTML is rendered.

**Reply by cderv** [reply, marked as answer]
I think this is currently expected for projects - this is the default behavior. You can opt-out by passing `--no-clean` to `quarto render` I believe
See `quarto render --help`
![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/9da95355-1a19-452b-8d41-0f95c6d1fd8d)

**Reply by fretwurst** [reply]
Can I change the default behavior of the Build Panel in the settings or key bindings? Or do I always have to type the command with this option?

**Reply by cderv** [reply]
Unfortunately this does not seem to be possible. This would be a feature request for https://github.com/rstudio/rstudio

**Comment by cderv** [comment]
> This naturally takes a considerable amount of time.

Regarding this, if the time is in computation execution, you should consider the options that Quarto offers at https://quarto.org/docs/projects/code-execution.html#freeze

If this is PDF rendering time from LaTeX, I can understand the trouble. 

However, for development, you should consider the preview workflow to avoid build the whole documents at each render
Example with book output: https://quarto.org/docs/books/#book-preview
